src/test/java/controllers/ProductControllerTest.java

```diff
@@ -0,0 +1,127 @@
+package controllers;
+
+import com.griddynamics.controllers.ProductController;
+import com.griddynamics.dto.ProductDTO;
+import com.griddynamics.service.ProductService;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ProductControllerTest {
+    private static final ProductDTO PRODUCT = new ProductDTO(1, "testName", 100, "testDesc", "testBrand", "img.png");
+
+    List<ProductDTO> products = Collections.singletonList(PRODUCT);
+
+    @InjectMocks
+    private ProductController productController;
+
+    @Mock
+    private ProductService productService;
+
+    @Test
+    public void getAllProductsShouldReturnNullTest() {
+        when(productService.findAll()).thenReturn(null);
```

Comment:
please write in following flow
it would be easier to read
method() {

//given
here you create local constants

//when
all whens

//then
call to test method + asserts + verifies

}